runners/direct-java/src/main/java/org/apache/beam/runners/direct/SplittableProcessElementsEvaluatorFactory.java

```diff
@@ -46,16 +50,32 @@
     implements TransformEvaluatorFactory {
   private final ParDoEvaluatorFactory<KeyedWorkItem<String, KV<InputT, RestrictionT>>, OutputT>
       delegateFactory;
-  private final EvaluationContext evaluationContext;
+  private final ScheduledExecutorService ses = Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder()
+                    .setThreadFactory(MoreExecutors.platformThreadFactory())
+                    .setNameFormat("direct-splittable-process-element-checkpoint-executor")
+                    .build());
 
   SplittableProcessElementsEvaluatorFactory(EvaluationContext evaluationContext) {
-    this.evaluationContext = evaluationContext;
     this.delegateFactory =
-        new ParDoEvaluatorFactory<>(
-            evaluationContext,
-            SplittableProcessElementsEvaluatorFactory
-                .<InputT, OutputT, RestrictionT>processFnRunnerFactory(),
-            ParDoEvaluatorFactory.basicDoFnCacheLoader());
+      new ParDoEvaluatorFactory<>(
+        evaluationContext,
+        SplittableProcessElementsEvaluatorFactory.
+          <InputT, OutputT, RestrictionT>processFnRunnerFactory(),
+          new CacheLoader<AppliedPTransform<?, ?, ?>, DoFnLifecycleManager>() {
+            @Override
+            public DoFnLifecycleManager load(final AppliedPTransform<?, ?, ?> application) {
+              if (!ProcessElements.class.isInstance(application.getTransform())) {
```

Comment:
`checkArgument`?

Your error message can also be simpler - `"%s is not a %s", application.getTransform(), ProcessElements.class.getSimpleName()`

Reply:
checkArgument -> we'll drop guava it seems so no need IMHO
for the error message: it is a dev message since it must not happen. "is not a" is too generic and doesnt give the why. Is it ok to keep it more regression prevention oriented?

Reply:
I don't believe we have any plan to stop using guava.

Developing either way requires delving into the actual code, but I don't think the exception message as is means anything - it's a basic precondition failure, and the message makes it sound like we've done work to figure out how to extract it when actually it's a much simpler problem.

Reply:
will move back to guava since the related thread (on dev@) didn't move enough so for now I agree with you on this one